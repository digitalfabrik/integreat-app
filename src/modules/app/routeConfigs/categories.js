// @flow

import RouteConfig from './RouteConfig'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import categoriesEndpoint from '../../endpoint/endpoints/categories'
import type { AllPayloadsType, GetLanguageChangePathParamsType, GetPageTitleParamsType } from './RouteConfig'
import CategoriesMapModel from '../../endpoint/models/CategoriesMapModel'
import Payload from '../../endpoint/Payload'
import CityModel from '../../endpoint/models/CityModel'

export type CategoriesRouteParamsType = {|city: string, language: string|}
type RequiredPayloadsType = {|categories: Payload<CategoriesMapModel>, cities: Payload<Array<CityModel>>|}

export const CATEGORIES_ROUTE = 'CATEGORIES'

const getCategoriesPath = ({city, language}: CategoriesRouteParamsType): string => `/${city}/${language}`

/**
 * CategoriesRoute, matches /augsburg/de*
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const categoriesRoute: Route = {
  path: '/:city/:language/:categoryPath*',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()
    const { city, language } = state.location.payload

    await fetchData(categoriesEndpoint, dispatch, state.categories, { city, language })
  }
}

const getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType =>
  ({categories: payloads.categoriesPayload, cities: payloads.citiesPayload})

const getLanguageChangePath = ({location, categories, language}: GetLanguageChangePathParamsType) => {
  const {city} = location.payload
  if (categories) {
    const category = categories.findCategoryByPath(location.pathname)
    if (category && category.id !== 0) {
      return category.availableLanguages.get(language) || null
    }
  }
  return getCategoriesPath({city, language})
}

const getPageTitle = ({t, categories, cityName, pathname}: GetPageTitleParamsType) => {
  const category = categories && categories.findCategoryByPath(pathname)
  return `${category && !category.isRoot() ? `${category.title} - ` : ''}${cityName}`
}

class CategoriesRouteConfig extends RouteConfig<CategoriesRouteParamsType, RequiredPayloadsType> {
  constructor () {
    super({
      name: CATEGORIES_ROUTE,
      getRequiredPayloads,
      route: categoriesRoute,
      getRoutePath: getCategoriesPath,
      getLanguageChangePath,
      getPageTitle
    })
  }
}

export default CategoriesRouteConfig
