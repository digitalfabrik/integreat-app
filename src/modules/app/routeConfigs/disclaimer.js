// @flow

import RouteConfig from './RouteConfig'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import disclaimerEndpoint from '../../endpoint/endpoints/disclaimer'
import type { GetLanguageChangePathParamsType, GetPageTitleParamsType } from './RouteConfig'

type DisclaimerRouteParamsType = {|city: string, language: string|}

export const DISCLAIMER_ROUTE = 'DISCLAIMER'

const getDisclaimerPath = ({city, language}: DisclaimerRouteParamsType): string =>
  `/${city}/${language}/disclaimer`

/**
 * DisclaimerRoute (for city specific disclaimers), matches /augsburg/de/disclaimer
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const disclaimerRoute: Route = {
  path: '/:city/:language/disclaimer',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()
    const {city, language} = state.location.payload

    await fetchData(disclaimerEndpoint, dispatch, state.disclaimer, {city, language})
  }
}

const getLanguageChangePath = ({location, language}: GetLanguageChangePathParamsType) =>
  getDisclaimerPath({city: location.payload.city, language})

const getPageTitle = ({t, cityName}: GetPageTitleParamsType) =>
  `${t('pageTitles.disclaimer')} - ${cityName}`

class DisclaimerRouteConfig extends RouteConfig<DisclaimerRouteParamsType> {
  constructor () {
    super({
      name: DISCLAIMER_ROUTE,
      route: disclaimerRoute,
      getRoutePath: getDisclaimerPath,
      getLanguageChangePath,
      getPageTitle
    })
  }
}

export default DisclaimerRouteConfig
