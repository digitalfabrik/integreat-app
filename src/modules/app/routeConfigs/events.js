// @flow

import React from 'react'
import type { AllPayloadsType, GetLanguageChangePathParamsType, GetPageTitleParamsType } from './types'
import Payload from '../../endpoint/Payload'
import EventModel from '../../endpoint/models/EventModel'
import RouteConfig from './RouteConfig'
import EventsPage from '../../../routes/events/containers/EventsPage'
import type { Route } from 'redux-first-router'

type RequiredPayloadType = {|events: Payload<Array<EventModel>>|}
type EventsRouteParamsType = {|city: string, language: string|}

export const EVENTS_ROUTE = 'EVENTS'

const getEventsPath = ({city, language}: EventsRouteParamsType): string => `/${city}/${language}/events`

/**
 * EventsRoute, matches /augsburg/de/events and /augsburg/de/events/begegnungscafe
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const eventsRoute: Route = '/:city/:language/events/:eventId?'

const renderPage = ({ events }: RequiredPayloadType) =>
  <EventsPage events={events.data} />

const getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadType =>
  ({ events: payloads.eventsPayload })

const getLanguageChangePath = ({location, events, language}: GetLanguageChangePathParamsType) => {
  const {city, eventId} = location.payload
  if (events && eventId) {
    const event = events.find(_event => _event.path === location.pathname)
    return (event && event.availableLanguages.get(language)) || null
  }
  return getEventsPath({city, language})
}

const getPageTitle = ({t, events, cityName, pathname}: GetPageTitleParamsType) => {
  const event = events && events.find(event => event.path === pathname)
  return `${event ? event.title : t('pageTitles.events')} - ${cityName}`
}

class EventsRouteConfig extends RouteConfig<RequiredPayloadType, EventsRouteParamsType> {
  constructor () {
    super({
      name: EVENTS_ROUTE,
      route: eventsRoute,
      getRoutePath: getEventsPath,
      getRequiredPayloads,
      getLanguageChangePath,
      getPageTitle
    })
  }
}

export default EventsRouteConfig
