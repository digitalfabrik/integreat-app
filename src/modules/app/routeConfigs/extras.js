// @flow

import RouteConfig from './RouteConfig'
import extrasEndpoint from '../../endpoint/endpoints/extras'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import type { GetLanguageChangePathParamsType, GetPageTitleParamsType } from './RouteConfig'

type ExtrasRouteParamsType = {|city: string, language: string|}

export const EXTRAS_ROUTE = 'EXTRAS'

const getExtrasPath = ({city, language}: ExtrasRouteParamsType): string =>
  `/${city}/${language}/extras`

/**
 * ExtrasRoute, matches /augsburg/de/extras and /augsburg/de/extras
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const extrasRoute: Route = {
  path: '/:city/:language/extras/:extraId?',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()
    const {city, language} = state.location.payload

    await fetchData(extrasEndpoint, dispatch, state.extras, {city, language})
  }
}

const getLanguageChangePath = ({location, language}: GetLanguageChangePathParamsType) =>
  getExtrasPath({city: location.payload.city, language})

const getPageTitle = ({t, cityName}: GetPageTitleParamsType) =>
  `${t('pageTitles.extras')} - ${cityName}`

class ExtrasRouteConfig extends RouteConfig<ExtrasRouteParamsType> {
  constructor () {
    super({
      name: EXTRAS_ROUTE,
      route: extrasRoute,
      getRoutePath: getExtrasPath,
      getLanguageChangePath,
      getPageTitle
    })
  }
}

export default ExtrasRouteConfig
