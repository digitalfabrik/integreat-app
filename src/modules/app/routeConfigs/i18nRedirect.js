// @flow

import RouteConfig from './RouteConfig'
import citiesEndpoint from '../../endpoint/endpoints/cities'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import Payload from '../../endpoint/Payload'
import CityModel from '../../endpoint/models/CityModel'
import type { AllPayloadsType } from './RouteConfig'

type I18nRedirectRouteParamsType = {|param?: string|}
type RequiredPayloadsType = {|cities: Payload<Array<CityModel>>|}

export const I18N_REDIRECT_ROUTE = 'I18N_REDIRECT'

const getI18nRedirectPath = ({param}: I18nRedirectRouteParamsType): string => `/${param || ''}`

/**
 * I18nRoute to redirect if no language is specified or to the not found route if the param is invalid.
 * Matches / and /param
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const i18nRedirectRoute: Route = {
  path: '/:param?',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()

    await fetchData(citiesEndpoint, dispatch, state.cities)
  }
}

const getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType => ({cities: payloads.citiesPayload})

class I18nRedirectRouteConfig extends RouteConfig<I18nRedirectRouteParamsType, RequiredPayloadsType> {
  constructor () {
    super({
      name: I18N_REDIRECT_ROUTE,
      route: i18nRedirectRoute,
      getRoutePath: getI18nRedirectPath,
      getPageTitle: () => '',
      getLanguageChangePath: () => null,
      getRequiredPayloads
    })
  }
}

export default I18nRedirectRouteConfig
