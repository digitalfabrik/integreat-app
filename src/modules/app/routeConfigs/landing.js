// @flow

import RouteConfig from './RouteConfig'
import citiesEndpoint from '../../endpoint/endpoints/cities'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import type { GetPageTitleParamsType } from './RouteConfig'

type LandingRouteParamsType = {|language: string|}

export const LANDING_ROUTE = 'LANDING'

const getLandingPath = ({language}: LandingRouteParamsType): string => `/landing/${language}`

/**
 * LandingRoute, matches /landing/de
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const landingRoute: Route = {
  path: '/landing/:language',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    await fetchData(citiesEndpoint, dispatch, getState().cities)
  }
}

const getPageTitle = ({t}: GetPageTitleParamsType) => t('pageTitles.landing')

class LandingRouteConfig extends RouteConfig<LandingRouteParamsType> {
  constructor () {
    super({
      name: LANDING_ROUTE,
      route: landingRoute,
      getRoutePath: getLandingPath,
      getPageTitle,
      getLanguageChangePath: () => null
    })
  }
}

export default LandingRouteConfig
