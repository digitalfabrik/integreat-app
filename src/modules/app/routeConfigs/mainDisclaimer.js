// @flow

import RouteConfig from './RouteConfig'
import { Route } from 'redux-first-router'
import type { GetPageTitleParamsType } from './RouteConfig'

export const MAIN_DISCLAIMER_ROUTE = 'MAIN_DISCLAIMER'

const getMainDisclaimerPath = (): string => '/disclaimer'

/**
 * MainDisclaimerRoute, matches /disclaimer
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const mainDisclaimerRoute: Route = '/disclaimer'

class MainDisclaimerRouteConfig extends RouteConfig<void> {
  constructor () {
    super({
      name: MAIN_DISCLAIMER_ROUTE,
      route: mainDisclaimerRoute,
      getRoutePath: getMainDisclaimerPath,
      getPageTitle: ({t}: GetPageTitleParamsType) => t('pageTitles.mainDisclaimer'),
      getLanguageChangePath: () => null
    })
  }
}

export default MainDisclaimerRouteConfig
