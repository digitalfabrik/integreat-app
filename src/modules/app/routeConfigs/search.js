// @flow

import RouteConfig from './RouteConfig'
import type { Dispatch, GetState, Route } from 'redux-first-router'
import fetchData from '../fetchData'
import categoriesEndpoint from '../../endpoint/endpoints/categories'
import type { AllPayloadsType, GetLanguageChangePathParamsType, GetPageTitleParamsType } from './RouteConfig'
import CityModel from '../../endpoint/models/CityModel'
import Payload from '../../endpoint/Payload'
import CategoriesMapModel from '../../endpoint/models/CategoriesMapModel'

type SearchRouteParamsType = {|city: string, language: string|}
type RequiredPayloadsType = {|categories: Payload<CategoriesMapModel>, cities: Payload<Array<CityModel>>|}

export const SEARCH_ROUTE = 'SEARCH'

const getSearchPath = ({city, language}: SearchRouteParamsType): string =>
  `/${city}/${language}/search`

/**
 * SearchRoute, matches /augsburg/de/search
 * @type {{path: string, thunk: function(Dispatch, GetState)}}
 */
const searchRoute: Route = {
  path: '/:city/:language/search',
  thunk: async (dispatch: Dispatch, getState: GetState) => {
    const state = getState()
    const {city, language} = state.location.payload

    await fetchData(categoriesEndpoint, dispatch, state.categories, {city, language})
  }
}

const getLanguageChangePath = ({location, language}: GetLanguageChangePathParamsType) =>
  getSearchPath({city: location.payload.city, language})

const getPageTitle = ({cityName, t}: GetPageTitleParamsType) =>
  `${t('pageTitles.search')} - ${cityName}`

const getRequiredPayloads = (payloads: AllPayloadsType): RequiredPayloadsType =>
  ({categories: payloads.categoriesPayload, cities: payloads.citiesPayload})

class SearchRouteConfig extends RouteConfig<SearchRouteParamsType, RequiredPayloadsType> {
  constructor () {
    super({
      name: SEARCH_ROUTE,
      route: searchRoute,
      getRoutePath: getSearchPath,
      getLanguageChangePath,
      getPageTitle,
      getRequiredPayloads
    })
  }
}

export default SearchRouteConfig
