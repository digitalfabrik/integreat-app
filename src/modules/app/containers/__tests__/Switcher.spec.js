import React from 'react'
import ConnectedSwitcher, { Switcher } from '../Switcher'
import Payload from '../../../endpoint/Payload'
import { shallow } from 'enzyme'
import { CATEGORIES_ROUTE } from '../../routes/categories'
import { LANDING_ROUTE } from '../../routes/landing'
import { MAIN_DISCLAIMER_ROUTE } from '../../routes/mainDisclaimer'
import { EXTRAS_ROUTE } from '../../routes/extras'
import { EVENTS_ROUTE } from '../../routes/events'
import { DISCLAIMER_ROUTE } from '../../routes/disclaimer'
import { SEARCH_ROUTE } from '../../routes/search'
import { I18N_REDIRECT_ROUTE } from '../../routes/i18nRedirect'
import configureMockStore from 'redux-mock-store'

describe('Switcher', () => {
  const errorPayload = new Payload(false, 'https://random.api.json', null, 'fake news')
  const dataPayload = new Payload(false, 'https://random.api.json', 'fetched fake news :D', null)
  const idlePayload = new Payload(false)
  const fetchingPayload = new Payload(true)

  const createSwitcher = ({currentRoute, citiesPayload = idlePayload, categoriesPayload = idlePayload,
    eventsPayload = idlePayload, extrasPayload = idlePayload,
    disclaimerPayload = idlePayload}) =>
    <Switcher viewportSmall={false} currentRoute={currentRoute} citiesPayload={citiesPayload}
              categoriesPayload={categoriesPayload} eventsPayload={eventsPayload} extrasPayload={extrasPayload}
              disclaimerPayload={disclaimerPayload} />

  describe('layout', () => {
    it('should render a location layout if the current route is a location layout route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: CATEGORIES_ROUTE})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('should render a layout with a footer if the current route is the landing route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: LANDING_ROUTE})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('should render a layout with a header and a footer as default', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: MAIN_DISCLAIMER_ROUTE})
      )

      expect(switcher).toMatchSnapshot()
    })
  })

  it('should return a spinner if the data has not been fetched yet', () => {
    expect(Switcher.renderFailureLoadingComponents(fetchingPayload)).toMatchSnapshot()
  })

  it('should return a failure if there was an error during fetching', () => {
    expect(Switcher.renderFailureLoadingComponents(errorPayload)).toMatchSnapshot()
  })

  describe('should get the right page if data has been fetched and', () => {
    it('is the categories route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: CATEGORIES_ROUTE, categoriesPayload: dataPayload, citiesPayload: dataPayload})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('is the extras route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: EXTRAS_ROUTE, extrasPayload: dataPayload, citiesPayload: dataPayload})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('is the events route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: EVENTS_ROUTE, eventsPayload: dataPayload, citiesPayload: dataPayload})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('is the disclaimer route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: DISCLAIMER_ROUTE, disclaimerPayload: dataPayload, citiesPayload: dataPayload})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('is the search route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: SEARCH_ROUTE, categoriesPayload: dataPayload, citiesPayload: dataPayload})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('is the main disclaimer route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: MAIN_DISCLAIMER_ROUTE})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('is the landing route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: LANDING_ROUTE, citiesPayload: dataPayload})
      )

      expect(switcher).toMatchSnapshot()
    })

    it('is the i18nRedirect route', () => {
      const switcher = shallow(
        createSwitcher({currentRoute: I18N_REDIRECT_ROUTE, citiesPayload: dataPayload})
      )

      expect(switcher).toMatchSnapshot()
    })
  })

  it('should map state to props', () => {
    const currentRoute = 'RANDOM_ROUTE'
    const location = {type: currentRoute, payload: {city: 'augsburg', language: 'de'}, prev: {payload: {param: 'param'}}}
    const mockStore = configureMockStore()
    const store = mockStore({
      location,
      events: fetchingPayload,
      cities: fetchingPayload,
      categories: fetchingPayload,
      disclaimer: fetchingPayload,
      extras: fetchingPayload,
      languages: dataPayload,
      viewport: {is: {small: true}}
    })

    const switcher = shallow(
      <ConnectedSwitcher store={store} />
    )

    expect(switcher.props()).toMatchObject({
      currentRoute,
      categoriesPayload: fetchingPayload,
      eventsPayload: fetchingPayload,
      extrasPayload: fetchingPayload,
      citiesPayload: fetchingPayload,
      disclaimerPayload: fetchingPayload,
      languages: 'fetched fake news :D',
      viewportSmall: true,
      city: 'augsburg',
      param: 'param',
      language: 'de'
    })
  })
})
