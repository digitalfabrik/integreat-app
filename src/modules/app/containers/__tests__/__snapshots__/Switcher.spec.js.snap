// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher layout should render a layout with a footer if the current route is the landing route 1`] = `
<Layout
  asideStickyTop={0}
  footer={<Translate(GeneralFooter) />}
>
  <LoadingSpinner />
</Layout>
`;

exports[`Switcher layout should render a layout with a header and a footer as default 1`] = `
<Layout
  asideStickyTop={0}
  footer={<Translate(GeneralFooter) />}
  header={
    <GeneralHeader
      viewportSmall={false}
    />
  }
>
  <MainDisclaimerPage />
</Layout>
`;

exports[`Switcher layout should render a location layout if the current route is a location layout route 1`] = `
<Connect(LocationLayout)>
  <LoadingSpinner />
</Connect(LocationLayout)>
`;

exports[`Switcher should get the right page if data has been fetched and is the categories route 1`] = `
<Connect(LocationLayout)>
  <Connect(CategoriesPage) />
</Connect(LocationLayout)>
`;

exports[`Switcher should get the right page if data has been fetched and is the disclaimer route 1`] = `
<Connect(LocationLayout)>
  <Connect(DisclaimerPage) />
</Connect(LocationLayout)>
`;

exports[`Switcher should get the right page if data has been fetched and is the events route 1`] = `
<Connect(LocationLayout)>
  <Connect(EventsPage) />
</Connect(LocationLayout)>
`;

exports[`Switcher should get the right page if data has been fetched and is the extras route 1`] = `
<Connect(LocationLayout)>
  <Connect(ExtrasPage) />
</Connect(LocationLayout)>
`;

exports[`Switcher should get the right page if data has been fetched and is the i18nRedirect route 1`] = `
<Layout
  asideStickyTop={0}
  footer={<Translate(GeneralFooter) />}
  header={
    <GeneralHeader
      viewportSmall={false}
    />
  }
>
  <Connect(I18nRedirect) />
</Layout>
`;

exports[`Switcher should get the right page if data has been fetched and is the landing route 1`] = `
<Layout
  asideStickyTop={0}
  footer={<Translate(GeneralFooter) />}
>
  <Connect(LandingPage) />
</Layout>
`;

exports[`Switcher should get the right page if data has been fetched and is the main disclaimer route 1`] = `
<Layout
  asideStickyTop={0}
  footer={<Translate(GeneralFooter) />}
  header={
    <GeneralHeader
      viewportSmall={false}
    />
  }
>
  <MainDisclaimerPage />
</Layout>
`;

exports[`Switcher should get the right page if data has been fetched and is the search route 1`] = `
<Connect(LocationLayout)>
  <Connect(SearchPage) />
</Connect(LocationLayout)>
`;

exports[`Switcher should return a failure if there was an error during fetching 1`] = `
<Translate(Failure)
  error="fake news"
/>
`;

exports[`Switcher should return a spinner if the data has not been fetched yet 1`] = `<LoadingSpinner />`;
